### ISSUE TITLE
Styling of dash_core_components

### ISSUE THREAD

**Issue by trhallam** [facultyai/dash-bootstrap-components#181]
Hi, I'm enjoying this package and generally finding it a breeze to integrate with the dash libraries but I had a question about applying the bootstrap styling the the dash_core_components.

In particular the Input component from dash (https://dash.plot.ly/dash-core-components/input)
has some extra options like 'step' for example which would be great for my app but the bootstrap ones don't seem to have this. I'm guessing this is because of how they are provided by bootstrap?

Is it possible to translate the bootstrap css styling to the dash tools if required?

Cheers
Tony

**Comment by tcbegley** [comment]
Hey Tony,

Glad you're enjoying *dash-bootstrap-components*! Thanks for raising this. I just submitted PR #183 which exposes additional props that *dash-core-components* makes available that we previously weren't. You can try it out by installing the prerelease I made to test it if you like:

```
pip install dash-bootstrap-components==0.4.1rc1
```

More generally, a lot of the Bootstrap styling is applied by CSS classes, and you can style *dash-core-components* simply by applying the relevant ones yourself (check [Bootstrap docs](https://getbootstrap.com/docs/4.3/components/forms/#form-controls) to figure out what they are). For example, in this case you could do

```python
dcc.Input(..., className="form-control")
```

to get a Bootstrap styled `dcc.Input`.

**Comment by trhallam** [comment]
Many thanks for the very prompt reply

Just tried both of those and whilst the `dcc.Input(...)` approach looks similiar its not quite the same. The *0.4.1rc1* package was perfect but, thank you.

I'm new to CSS and Bootstrap so still working it out as I go, I have worked through most of those Bootstrap docs but it is not always clear to me where Bootstrap styling calls should go when creating components from `dcc` or `dbc`. 

Sometimes there appears to be specific keyword arguments in a `stlye` dictionary and other times it belongs to the `className` keyword as a full string. Perhaps I am missing something in the docs?

**Comment by tcbegley** [comment]
I often find the inspector in the browser to be very helpful when understanding what the different style arguments and classes are doing. For example, inspecting the "Components" button on the *dash-bootstrap-components* docs homepage I can see this sort of thing:

![image](https://user-images.githubusercontent.com/15220906/56202732-9b848100-603b-11e9-86bb-6c17ed480d1f.png)

I can see what `color`, `background-color` and `border-color` are being set as, and that it's because the button has the `.btn-primary` class applied. You can turn these styles on and off individually, or add new styles in the inspector (which are reset when you refresh) to experiment and get the look right. Then when you're happy, copy the styles across to the code.

It's sometimes also useful to look at Bootstrap CSS itself and see what the different classes are actually doing. For example, here's the definition of the form control class: https://github.com/twbs/bootstrap/blob/master/dist/css/bootstrap.css#L1705-L1719

It's one of those things you just need to keep playing around with and it will slowly start to make sense.

**Comment by trhallam** [comment]
I have seen the inspector before but totally forgot about it. That is by far the most useful styling tip I've seen for playing with `dash`. Thank you! I was trying to read the src to figure out `DashTable` and the inspector is far more informative. As with anything it helps when you know where to start.

**Comment by tcbegley** [comment]
This is now available in version 0.5.0, no more need for prereleases 🎉
